src/rdkafka.c

```diff
@@ -3181,33 +3202,79 @@ rd_kafka_message_t *rd_kafka_consumer_poll(rd_kafka_t *rk, int timeout_ms) {
 }
 
 
-rd_kafka_resp_err_t rd_kafka_consumer_close(rd_kafka_t *rk) {
+/**
+ * @brief Consumer close.
+ *
+ * @param rkq The consumer group queue will be forwarded to this queue, which
+ *            which must be served (rebalance events) by the application/caller
+ *            until rd_kafka_consumer_closed() returns true.
+ *            If the consumer is not in a joined state, no rebalance events
+ *            will be emitted.
+ */
+static rd_kafka_error_t *rd_kafka_consumer_close_q(rd_kafka_t *rk,
+                                                   rd_kafka_q_t *rkq) {
         rd_kafka_cgrp_t *rkcg;
-        rd_kafka_op_t *rko;
-        rd_kafka_resp_err_t err = RD_KAFKA_RESP_ERR__TIMED_OUT;
-        rd_kafka_q_t *rkq;
+        rd_kafka_error_t *error = NULL;
 
         if (!(rkcg = rd_kafka_cgrp_get(rk)))
-                return RD_KAFKA_RESP_ERR__UNKNOWN_GROUP;
+                return rd_kafka_error_new(RD_KAFKA_RESP_ERR__UNKNOWN_GROUP,
```

Comment:
i think i would err on the side of making this a noop (mark consumer as closed and return) vs an error.

you could argue that the more constrained behavior -> informative, but I feel like if there is a method as generic as close, on a class, it generally shouldn't be an error to call it.